src/Raven.Server/Documents/Indexes/IndexNameFinder.cs

```diff
@@ -37,13 +37,13 @@ private static string FindName(string[] collections, IReadOnlyCollection<IndexFi
             
             var combinedFields = string.Join("And", fields.Select(x => IndexField.ReplaceInvalidCharactersInFieldName(x.Name)).OrderBy(x => x));
```

Comment:
Okay, that is much better. That said, we do a lot of allocations there for the `explain` method, even if it isn't used (just to create the lambda to pass)